### DISCUSSION TITLE
Is there a way to declare "allowed values" for a property?

### DISCUSSION THREAD

**Discussion by mstenta** [json-schema-org/community#51] [Q&A]
Let's say we have a JSON object like this:

```
{
  "id": 1,
  "name": "My object",
  "type": "foo",
}
```

And the "allowed values" of the `type` property are either `foo` or `bar`. Is there any way to describe this with JSON Schema?

**Comment by gbathree** [comment]
Looking here: https://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties

Let's take this as an example.  Here, it appears by using `enum` we are specifying the allowed values... 

```
{
  "type": "object",
  "properties": {
    "number": { "type": "number" },
    "street_name": { "type": "string" },
    "street_type": { "enum": ["Street", "Avenue", "Boulevard"] }
  },
  "additionalProperties": false
}
```

Now, one could image in **Plant Data Service* land the allowed values could get quite long and complex.  Perhaps what you want is to a way to specify `"enum":["a", "b", "c"]` where a/b/c is a reference to a big object.  Then you could run a program to 'compile' the actual things together... or perhaps (better yet) we can use a HTML reference in enum so it's web-based and version controlled... I'd like to brainstorm this with @sudokita also...
